src/engine.rs

```diff
@@ -104,18 +104,27 @@ impl Engine {
             let epv = end_point_voices.clone();
 
             let sounds = Arc::new(Mutex::new(Vec::new()));
-            future_to_exec = Some(stream.for_each(move |mut buffer| -> Result<_, ()> {
+            future_to_exec = Some(stream.for_each(move |mut buffer| -> Result<(), ()> {
                 let mut sounds = sounds.lock().unwrap();
 
                 {
                     let mut pending = epv.pending_sounds.lock().unwrap();
                     sounds.append(&mut pending);
                 }
 
+
                 if sounds.len() == 0 {
                     return Ok(());
                 }
                 sounds.retain(|s| !s.1.local_dead);
```

Comment:
I'd recommend merging these two calls to retain, we can take advantage of some short circuiting optimization if we do.